Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm

```diff
@@ -1189,7 +1189,20 @@
             endPosition = nextPosition;
     }
 
-    return makeSimpleRange(contextStartPosition, endPosition);
+    // Strip trailing newlines.
+
+    auto finalRange = makeSimpleRange(contextStartPosition, endPosition);
+    if (!finalRange)
+        return std::nullopt;
+
+    auto text = WebCore::plainText(*finalRange);
+    while (text.endsWith('\n')) {
+        endPosition = WebCore::positionOfNextBoundaryOfGranularity(endPosition, WebCore::TextGranularity::CharacterGranularity, WebCore::SelectionDirection::Backward);
+        finalRange = makeSimpleRange(contextStartPosition, endPosition);
+        text = WebCore::plainText(*finalRange);
```

Comment:
I believe it's necessary for the loop to work properly, in the case where there are multiple trailing newlines, since it needs to be recomputed after adjusting the range